projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.e2e.ts

```diff
@@ -411,27 +409,42 @@ describe('dot-key-value', () => {
     describe('@Methods', () => {
         beforeEach(async () => {
             element.setAttribute('name', 'fieldName');
-            element.setAttribute('required', true);
+            element.setAttribute('value', 'first key|first value,second key|second value');
+
             spyValueChangeEvent = await page.spyOnEvent('valueChange');
             spyStatusChangeEvent = await page.spyOnEvent('statusChange');
         });
 
         describe('reset', () => {
-            it('should clear the field and emit', async () => {
-                element.setAttribute('value', 'first key|first value,second key|second value');
+            it('should clear the field and emit invalid (field required)', async () => {
+                element.setAttribute('required', true);
+                await page.waitForChanges();
+
+                element.callMethod('reset');
+                await page.waitForChanges();
+
+                expect(spyValueChangeEvent).toHaveReceivedEventDetail({
+                    name: 'fieldName',
+                    value: ''
+                });
+                expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                    name: 'fieldName',
+                    status: { dotPristine: true, dotTouched: false, dotValid: false }
+                });
+            });
+            it('should clear the field and emit valid (field not required)', async () => {
                 await page.waitForChanges();
```

Comment:
is this needed?

Reply:
Yes, because in line 411 and 412 (inside `beforeEach`) I'm doing `setAttribute` and not `waitForChanges` after.